### ISSUE TITLE
Please support Bahasa Indonesia

### ISSUE THREAD

**Issue by ideatrap** [niedev/RTranslator#70]
There are 275.5 million population in Indonesia. Can you support Bahasa Indonesia?

**Comment by niedev** [comment]
Hi, NLLB doesn't support this language, but in the future I will integrate new models for translation, when this language will be supported I will let you know.

**Comment by ideatrap** [comment]
Really appreciate. Indonesia a booming economy, an has beautiful Bali. That will benefit many people traveling to Indonesia.

**Comment by ideatrap** [comment]
Looking forward to it

**Comment by drajabr** [comment]
+1 I was disappointed that after hours of searching I found the app that actually works offline with real on-device voice recognition, but Indonesian Language isn't on the list! :( 

@ideatrap  Did you find any other app that support Indonesian voice translation offline? This will be a literally life saver for me.

**Comment by drajabr** [comment]
@niedev Isn't it ironic that NLLB stands for "No Language Left Behind" but didn't include the language of about 300 million person on earth xD 

BTW, I'm not sure if this is the same model you use, but I see on their page now Indonesian language is supported! Any update?

**Comment by niedev** [comment]
@drajabr @ideatrap yeah sorry, I said it is not supported but I was wrong, in theory it is supported but I tested it some time ago and the translation with NLLB with Indonesian does not work, it treats it as if it were English, I think it is a bug in the model, I can not do anything about it sorry, and yes, it is a shame

**Comment by drajabr** [comment]
@niedev With all due respect, that is not true, it correctly detects the language and do translate it properly. 

I'm sorry, but if an amateur computer enthusiastic person like me could give it a quick test and confirm the model detection and translation for the Indonesian language without issue, then you probably shouldn't throw the `I think it is a bug in the model` sentence, and completely ignores the possibility that it is a real issue needs to be investigated.

**Comment by niedev** [comment]
@drajabr Let's clarify something, this is a free and open source project, I don't work for you, and this is not a customer service. So if you think I'm wrong, that's ok, tell me that I'm wrong and I will investigate further, but I don't owe it to you. So keep the lecture to yourself and show some respect.
That said, from the tests I did some time ago, the language had problems and still has the same problems, I don't know where you did the test, but with this model, with this quantization, Indonesian is treated like english, the only exception is with using beam search (with the beam size = 4) but I can't insert a language that only works when some app settings (which I will add soon, but as a beta feature) have a certain value.

![Image](https://github.com/user-attachments/assets/eba8db9d-e6cb-4622-8dbe-ac7e24a97c3c)

**Comment by drajabr** [comment]
That's fair, I am sorry for how my reply went way out of context.

And now I see that you used the "stripped" version of the model, if its valid to say that, and probably that's why it worked for me but it doesn't work in here.
I have no knowledge in all these details and I shouldn't have commented with that confidence before educating myself a little bit about your amazing project.
With all respect and support, I hope you accept my sincere apology. 

Maybe you can reopen the issue so we get notified when there's an update about it? 
And if there's a beta release or sorta I can provide you with testing almost 24/7, if that helps.

**Comment by niedev** [comment]
Thank you for your sincere apology and understanding, I appreciate it. I can reopen the issue and let you know if there is any news, but it is very likely that the support for this language will not come soon, since there is not much I can do at the moment, but when I will integrate other models into the app other than NLLB if this language will be supported I will let you know.

**Comment by QiubyZ** [comment]
Okay, so far there is no Indonesian language.

**Comment by niedev** [comment]
> Okay, so far there is no Indonesian language.

Yes, have you read the previous comments?

**Comment by zzw8587** [comment]
I was once very curious as to why they even supported Javanese language but didn't have Indonesian language. It felt extremely strange. It turned out that Indonesian language was actually recognized as English.
A while ago, when I was climbing a volcano in Indonesia, in an area with no signal and the tour guide couldn't speak English, I felt extremely in need of a large offline translation model. Hahaha
I hope the author can support the Indonesian language in the future.

**Comment by drajabr** [comment]
> when I was climbing a volcano in Indonesia, in an area with no signal and the tour guide couldn't speak English, I felt extremely in need of a large offline translation model. Hahaha I hope the author can support the Indonesian language in the future.

Don't climb volcanos 😜 


No, I'm joking.

Sadly, I ended up using google translate, offline support text only though which sucks.

**Comment by Peanutje** [comment]
Hi!
Is there any update on adding Bahasa Indonesia to the app? Is there any way that i can do this myself? I really hate using Google bc of privacy issues. 
Did anybody work on this in the meantime? Or could anyone provide me with the information trying to add it myself for the community? 

We need the support for Indonesia asap please. Thank you in advance.

**Comment by niedev** [comment]
Hi @Peanutje,

No, you can't do it by yourself. If a language isn't supported by the translation model, you would need a lot of money and expertise to add a language to it. The good news is that the new 3.0 version will support new models, and some of them support Indonesian, so you just need to wait for the 3.0 version release!

**Comment by niedev** [comment]
The [first beta](https://github.com/niedev/RTranslator/releases/tag/3.0.0-beta1) of the new 3.0 version is out! All the new models support Indonesian, let me know if they work well.
